### DISCUSSION TITLE
iPadOS: server URL isn’t pasted & unable to be clicked

### DISCUSSION THREAD

**Discussion by jenkshields** [community/community#26910] [Codespaces]
<p>Kia ora!</p>
<p>I’m so excited by the Codespaces beta! I’ve just started playing around, testing it out with a Next.js website I built on the iPad recently (using Play.js - but so excited for the possibilities Codespaces provides!)</p>
<p>I’m running into an issue with forwarded ports - in the terminal output, I’m unable to click on the localhost URL using the touch interface - I see the “follow link” tooltip, but without a physical keyboard I can’t use cmd + click. When clicking to copy the server URL in the forwarded ports panel, I see the notification that the URL has been copied to the clipboard, but this doesn’t seem to actually happen and I’m not able to paste it! These two things mean that I don’t think there’s any way to view my web app - which is obviously not ideal!</p>
<p>Has anyone else run into this, or been able to resolve it?</p>

**Comment by Chuxel** [comment]
<p>Another way you can get to the forwarded URLs is as follows:</p>
<ol>
<li>Go to the “Remote Explorer”<br>
<img src="https://user-images.githubusercontent.com/312252/181100582-5da724c5-bf21-4a67-bcef-e668b9d8ad1d.png" alt="image" data-base62-sha1="gpNZv74SluPFtODRY8uQsWUgrHx" width="182" height="240">
</li>
<li>Look under “Codespace Details” and you should see the port listed<br>
<img src="https://user-images.githubusercontent.com/312252/181100583-d2d37153-810e-40d4-bdc9-b0fa77b99262.png" alt="image" data-base62-sha1="vcYxDJWyE3M9owhIjPnXk0MJFaF" width="610" height="290">
</li>
<li>Click / touch it and the forwarded url should open!</li>
</ol>

**Comment by jenkshields** [comment]
<p>Thanks for this! Unfortunately, touching it doesn’t open the link, just brings up the option icons!</p>

**Comment by lostintangent** [comment, marked as answer]
<p>I believe this may be due to pop-ups being silently blocked. If you go into your Safari settings, and disable the <code>Block pop-ups</code> option, are you able to click on the port to open it?</p>
<p>As an aside: if you run <code>npx -q codespaces-port &lt;port&gt;</code> from the integrated terminal, that will print out the URL for the port that you could open in a new tab.</p>

**Comment by jenkshields** [comment]
<p>Thanks so much for this - allowing pop ups got it working!</p>

**Comment by lostintangent** [comment]
<p>Awesome! Thanks for confirming that :+1:</p>
